core-rust/state-manager/src/staging/result.rs

```diff
@@ -247,11 +247,13 @@ impl ProcessedCommitResult {
                                 new: value.clone(),
                                 previous,
                             }),
-                            Some(_) => None, /* Same value as before, ignore */
-                            None => Some(ChangeAction::Create(value.clone())),
+                            Some(_) => None, /* Same value as before (i.e. not updated), ignore */
+                            None => Some(ChangeAction::Create { new: value.clone() }),
                         }
                     }
-                    DatabaseUpdate::Delete => Some(ChangeAction::Delete),
+                    DatabaseUpdate::Delete => store
+                        .get_substate(&partition_key, &sort_key)
+                        .map(|previous| ChangeAction::Delete { previous }),
```

Comment:
_(note from author)_

This has the same behavior as the "previous value same" for the Update above:
If the value did not exist, then the Delete is ignored.